HotelApp/Controllers/HotelController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Threading.Tasks;
+using HotelApp.API.DbContexts.Repositories;
+using HotelApp.API.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace HotelApp.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class HotelController : ControllerBase
+    {
+        private readonly IHotelRepository _hotelRepository;
+
+        public HotelController(IHotelRepository hotelRepository)
+        {
+            _hotelRepository = hotelRepository;
+        }
+
+        [HttpPost]
+        [Authorize(Roles = "Registered user")]
+        public async Task<IActionResult> RegisterHotel([FromBody] RegisterHotelDTO model)
+        {
+            await _hotelRepository.CreateHotelAsync(model);
+
+            return Ok(new ResponseDTO
+            {
+                Status = "Success",
+                Message = "Hotel registered successfully!"
+            });
```

Comment:
Nešto što sam mogao prije reć, kod Create akcija je dobro vratiti Id entiteta, tak da onaj ko je kreirao odmah može dohvatiti detalje nakon create akcije